api/app/controllers/api/v1/offers_controller.rb

```diff
@@ -1,31 +1,396 @@
+      offer.update_attributes!(status: "Unsent", accept_date: nil, commentary: nil, hr_status: nil, nag_count: nil,
+                               print_time: nil, send_date: nil, signature: nil)
+      offer = offer.format
+      render status: 200, json: {message: "You've changed the status of #{offer[:applicant][:first_name]} #{offer[:applicant][:last_name]}'s offer for #{offer[:position]} to 'Unsent'."}
+    end
+
+    private
+    def get_contract_html(offer, office_template=false)
+      contract_dir = "#{Rails.root}/app/views/contracts/#{ENV["CONTRACT_SUBDIR"]}"
+      # load the offer as a Liquid template
+      if office_template
+        template = Liquid::Template.parse(File.read("#{contract_dir}/offer-template-office.html"))
+      else
+        template = Liquid::Template.parse(File.read("#{contract_dir}/offer-template.html"))
+      end
+      # font.css and header.css contain base64-encoded data since we need all
+      # data to be embedded in the HTML document
+      styles = { "style_font" => File.read("#{contract_dir}/font.css"),
+                 "style_header" =>  File.read("#{contract_dir}/header.css")}
+      subs = offer.format.merge(styles)
+      rendered = template.render(subs)
+    end
+
+    def offer_params
+      params.permit(:hr_status, :ddah_status, :commentary, :hours)
+    end
+
+    def get_all_offers(offers)
+      offers.map do |offer|
+        offer.format
+      end
+    end
+
+    def update_print_status(offer_id)
+      offer = Offer.find(offer_id)
+      if offer[:hr_status] == "Processed"
+        offer.update_attributes!({print_time: DateTime.now})
+      else
+        offer.update_attributes!({hr_status: "Printed", print_time: DateTime.now})
+      end
+    end
+
+    '''
+      Sets the status of an offer to either `Accepted`, `Rejected`, or `Withdrawn`
+    '''
+    def status_setter(params)
+      status = get_status(params)
+      offer = Offer.find(params[:offer_id])
+      if status[:action] == "withdraw" || offer[:status] != "Unsent"
+        update_status(offer, status)
+        render json: {success: true, status: status[:name].downcase, message: "You've just #{status[:name].downcase} this offer."}
+      else
+        render status: 404, json: {success: false, message: "You cannot #{status[:action]} an unsent offer."}
+      end
+    end
+
+    def update_status(offer, status)
+      if status[:action]=="accept"
+        offer.update_attributes!({status: status[:name], signature: status[:signature], accept_date: DateTime.now})
+      else
+        offer.update_attributes!({status: status[:name]})
+      end
+    end
+
+    def get_status(code)
+      case params[:status]
+      when "accept"
+        return {name: "Accepted", action: "accept", signature: params[:signature]}
+      when "reject"
+        return {name: "Rejected", action: "reject"}
+      when "withdraw"
+        return {name: "Withdrawn", action: "withdraw"}
+      end
+    end
+
+    def set_domain
```

Comment:
This environment variable should be set at the application level through docker, not in this controller.